### PR TITLE
MSL-357: Jenkins Slave AMI Profile and added nodejs role

### PR DESCRIPTION
**Pull request by rohitagarwal01**

### CODE REVIEW

File: roles/jenkins-slave/vars/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+
+jenkins_dns: build.kenzan-devops.com
```
Comment (dpeterka) [inline roles/jenkins-slave/vars/main.yml]: Do we want this to be so specific to our setup? Perhaps a placeholder or global requirement to pass in something like -var=mydomain.net

File: roles/nodejs/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+
+# Include OS-specific installation tasks.
+- include: setup-RedHat.yml
+  when: ansible_os_family == 'RedHat'
+
+- include: setup-Debian.yml
+  when: ansible_os_family == 'Debian'
+
+- name: npm disable self signed cert fail
```
Comment (dpeterka) [inline roles/nodejs/tasks/main.yml]: This can be removed

File: roles/jenkins-slave/tasks/setup-RedHat.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+
+- name: Install Packages
+  package: name={{item}} state=latest enablerepo=epel
+  with_items:
+    - git
+    - ruby19
```
Comment (dpeterka) [inline roles/jenkins-slave/tasks/setup-RedHat.yml]: Should be on Ruby 2.0

File: roles/jenkins-slave/tasks/setup-RedHat.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    - rpmdevtools
+    - python-pip
+    - ruby19-devel
```
Comment (dpeterka) [inline roles/jenkins-slave/tasks/setup-RedHat.yml]: Should be on Ruby 2.0

File: roles/jenkins-slave/tasks/setup-RedHat.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    - GeoIP-devel
+    - gd-devel
```
Comment (dpeterka) [inline roles/jenkins-slave/tasks/setup-RedHat.yml]: Remove

File: roles/jenkins-slave/tasks/setup-RedHat.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    - gperftools-devel
+    - libxslt-devel
+    - openssl-devel
+    - pcre-devel
+    - perl-devel
+    - zlib-devel
```
Comment (dpeterka) [inline roles/jenkins-slave/tasks/setup-RedHat.yml]: Doubt we need this.

Comment (dpeterka) [inline roles/jenkins-slave/tasks/setup-RedHat.yml]: Doubt we need this.

File: roles/jenkins-slave/tasks/setup-RedHat.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    - rpmdevtools
+    - python-pip
+    - ruby19-devel
+    - GeoIP-devel
+    - gd-devel
+    - gperftools-devel
+    - libxslt-devel
+    - openssl-devel
+    - pcre-devel
+    - perl-devel
```
Comment (dpeterka) [inline roles/jenkins-slave/tasks/setup-RedHat.yml]: Doubt we need this.

File: roles/jenkins-slave/tasks/setup-RedHat.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    - ruby19
+    - rpmdevtools
+    - python-pip
+    - ruby19-devel
+    - GeoIP-devel
+    - gd-devel
+    - gperftools-devel
+    - libxslt-devel
+    - openssl-devel
+    - pcre-devel
```
Comment (dpeterka) [inline roles/jenkins-slave/tasks/setup-RedHat.yml]: Doubt we need this.

**Comment by rohitagarwal01** [comment]
Hi @dpeterka, can you review updates, will squash commits and put in new PR if all is good

Thanks & Regards,
Rohit

**Comment by dpeterka** [comment]
Should use upstart and make this work on either ubuntu or redhat

**Comment by rohitagarwal01** [comment]
Closing PR, will put in a new one

**Pull request closed without merging**